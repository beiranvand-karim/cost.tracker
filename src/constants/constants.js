export const NO_FILTER = "NO_FILTER";
export const DUMMY_ACTION = "DUMMY_ACTION";
