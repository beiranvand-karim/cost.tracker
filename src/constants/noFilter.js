const NO_FILTER = 'NO_FILTER';

export default NO_FILTER;
