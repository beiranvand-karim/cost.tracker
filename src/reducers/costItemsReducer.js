import {
  ADD_COST_ITEM,
  ADD_FILTERED_COST_ITEMS
} from "../actions/costItemsActions";

const defaultState = {
  costItems: []
};

export default function costItemsReducer(state = defaultState, action) {
  switch (action.type) {
    case ADD_COST_ITEM:
      const { costItems } = state;
      costItems.unshift(action.payload);
      return {
        ...state,
        costItems
      };
    case ADD_FILTERED_COST_ITEMS:
      return {
        ...state,
        costItems: action.payload
      };
    default:
      return state;
  }
}
