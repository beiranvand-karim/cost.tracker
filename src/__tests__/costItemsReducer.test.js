/* global expect, describe, it, beforeEach */
import costItemsReducer from "../reducers/costItemsReducer";
import { DUMMY_ACTION } from "../constants/constants";
import costItems from "../__helpers__/costItems";
import {
  ADD_COST_ITEM,
  ADD_FILTERED_COST_ITEMS
} from "../actions/costItemsActions";

describe("costItemsReducer", () => {
  it("should matches the initial state", () => {
    const action = { type: DUMMY_ACTION };
    expect(costItemsReducer(undefined, action)).toMatchSnapshot();
  });

  it("should matches the add filtered cost items state", () => {
    const action = {
      type: ADD_FILTERED_COST_ITEMS,
      payload: costItems
    };
    expect(costItemsReducer(undefined, action)).toMatchSnapshot();
  });

  it("should matches the add cost item state", () => {
    const action = {
      type: ADD_COST_ITEM,
      payload: costItems[0]
    };
    expect(costItemsReducer(undefined, action)).toMatchSnapshot();
  });
});
