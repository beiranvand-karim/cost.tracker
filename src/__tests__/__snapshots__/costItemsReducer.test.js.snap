// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`costItemsReducer should matches the add cost item state 1`] = `
Object {
  "costItems": Array [
    Object {
      "category": "CLOTHING",
      "comment": "this is a comment",
      "cost": 1000,
    },
  ],
}
`;

exports[`costItemsReducer should matches the add filtered cost items state 1`] = `
Object {
  "costItems": Array [
    Object {
      "category": "CLOTHING",
      "comment": "this is a comment",
      "cost": 1000,
    },
    Object {
      "category": "RENT",
      "comment": "this is a comment",
      "cost": 1500,
    },
    Object {
      "category": "FOOD",
      "comment": "this is a comment",
      "cost": 2000,
    },
  ],
}
`;

exports[`costItemsReducer should matches the initial state 1`] = `
Object {
  "costItems": Array [],
}
`;
